test/unit_test/passes/pytorch/test_torch_trt_conversion.py

```diff
@@ -0,0 +1,85 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import sys
+import tempfile
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+import torch
+
+import olive.passes.pytorch.sparsegpt_utils as sparsegpt_utils
+from olive.data.template import huggingface_data_config_template
+from olive.model import PyTorchModel
+from olive.passes.olive_pass import create_pass_from_dict
+from olive.passes.pytorch import TorchTRTConversion
+from olive.systems.local import LocalSystem
+
+
+class MockTRTLinearLayer(torch.nn.Module):
+    pass
+
+
+def mocked_torch_zeros(*args, **kwargs):
+    if "device" in kwargs:
+        del kwargs["device"]
+    return torch.ones(*args, **kwargs) * 0
+
+
+# all the following patches are needed to run the test on CPU
+# make torch.cuda.is_available return True
+@patch("torch.cuda.is_available", return_value=True)
+# make tensor_data_to_device return the input tensor, ignore device
+@patch("olive.common.utils.tensor_data_to_device", side_effect=lambda x, y: x)
+# make torch.nn.Module.to do nothing
+@patch("torch.nn.Module.to", side_effect=lambda device: None)
+# replace device in kwargs with "cpu"
+@patch("torch.zeros", side_effect=mocked_torch_zeros)
+def test_sparsegpt(mock_is_cuda_available, mock_tensor_data_to_device, mock_module_to, mock_torch_zeros):
+    # setup
+    # mock trt utils since we don't have tensorrt and torch-tensorrt installed
+    mock_trt_utils = MagicMock()
+    mock_compile_trt_model = MagicMock()
+    mock_compile_trt_model.return_value = MockTRTLinearLayer()
+    mock_trt_utils.compile_trt_model = mock_compile_trt_model
+
+    sys.modules["olive.passes.pytorch.trt_utils"] = mock_trt_utils
```

Comment:
>sys.modules["olive.passes.pytorch.trt_utils"] = mock_trt_utils [](http://example.com/codeflow?start=4&length=62)

I suppose this line is to avoid import TensorRT related stuff in UT. right?

Reply:
yes, tenssort requires gpu, etc